server/src/index.ts

```diff
@@ -51,6 +55,33 @@ env.config({ path: path.resolve(process.cwd(), '..', '.env'), example: path.reso
     }
   }))
 
+  // OAUTH 2
+
+  const key = fs.readFileSync(path.join(__dirname, '/cert/localhost-key.pem'))
+  const cert = fs.readFileSync(path.join(__dirname, '/cert/localhost.pem'))
+
+  const server = https.createServer({ key: key, cert: cert }, app)
+
+  app.use(passport.initialize())
+  app.use(passport.session())
+
+  passport.use('oauth2', oauthStrategy)
+
+  passport.serializeUser(async (user: any, done: any) => serializeUser(user, done))
+
+  passport.deserializeUser(async (id: string, done: any) => deserializeUser(id, done))
+
+  app.get('/api/login', passport.authenticate('oauth2'))
+
+  app.get('/api/callback', passport.authenticate('oauth2'), (req: express.Request, res: express.Response) => {
+    // @ts-ignore is valid
```

Comment:
Hier bitte einmal den Type ergänzen in unserer types.ts. Dort könntest du statt
```ts
req: Request & { session: Session }
```
schreiben
```ts
req: Request & { session: Session & { passport: { user: User }}}
```